### ISSUE TITLE
Python windows build problems

### ISSUE THREAD

**Issue by a4z** [cross-language-cpp/djinni-support-lib#40]
Even after the change which targets issue #32 , and fixes that , there are several issues with the Windows build for Python.

Then, the problems: 

## First
https://github.com/cross-language-cpp/djinni-support-lib/blob/755d1652d4c5ff84cfe08a275c808efefe27c751/test-suite/CMakeLists.txt#L239

On Windows, the PYTHON_PATH needs to be separated with a `;` character, not with a `:`

## Second
Even when this is done, when it is used on Windows
https://github.com/cross-language-cpp/djinni-support-lib/blob/755d1652d4c5ff84cfe08a275c808efefe27c751/test-suite/CMakeLists.txt#L248
CMake replaces the semicolon with a space, so something must be done to prevent that from happen,

## Third
When this is fixed, on Windows with MSVC, the required created lib (mylib) needs to be found, what is not the case if this library gets generated into subfolder, like Debug. 
Just output them into the current binary dir would work, but would also cause other side effects, like, problems with everything that support a multiconfig build (msvc, ninja, ...)

## Forth
And when this is fixed and we would have a custom_target command that works, a next problem occurse. 
```
opt = compiler.runtime_library_dir_option(dir)
  File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\_msvccompiler.py", line 519, in runtime_library_dir_option
    raise DistutilsPlatformError(
distutils.errors.DistutilsPlatformError: don't know how to set runtime library search path for MSVC
Batch file failed at line 3 with errorcode 1
```

## Actions
Due to time restrictions I can not say how long it will take me to dig into all that, so help would be welcome.
Even if not the whole issue gets fixed, but parts from it.

## First step
As a first step  I suggest: 
Disable it in CMake and bring a nice message, instead of letting users run into cryptic errors.

**Comment by a4z** [comment]
problem 1,2,3 have been solved in this dev branch https://github.com/a4z/djinni-support-lib/tree/private/moreWindowsTweaks

still, there is that problem

```
....
  Building CFFI lib
  PyCFFIlib_cffi.c
CUSTOMBUILD : warning : I don't know what to do with 'runtime_library_dirs': ['.'] [C:\Users\a4z\work\djinni\support-lib\build\py2\test-suite\PyCFFIlib_cffi.vcxproj]
  Traceback (most recent call last):
    File "C:\Users\a4z\work\djinni\support-lib\build\py2\test-suite\generated-src\cffi\pycffi_lib_build.py", line 25, in <module>
      ffi.compile()
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\site-packages\cffi\api.py", line 725, in compile
      return recompile(self, module_name, source, tmpdir=tmpdir,
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\site-packages\cffi\recompiler.py", line 1564, in recompile
      outputfilename = ffiplatform.compile('.', ext,
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\site-packages\cffi\ffiplatform.py", line 22, in compile
      outputfilename = _build(tmpdir, ext, compiler_verbose, debug)
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\site-packages\cffi\ffiplatform.py", line 51, in _build
      dist.run_command('build_ext')
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\site-packages\setuptools\command\build_ext.py", line 79, in run
      _build_ext.run(self)
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\command\build_ext.py", line 340, in run
      self.build_extensions()
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\command\build_ext.py", line 449, in build_extensions
      self._build_extensions_serial()
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\command\build_ext.py", line 474, in _build_extensions_serial
      self.build_extension(ext)
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\site-packages\setuptools\command\build_ext.py", line 196, in build_extension
      _build_ext.build_extension(self, ext)
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\command\build_ext.py", line 551, in build_extension
      self.compiler.link_shared_object(
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\ccompiler.py", line 713, in link_shared_object
      self.link(CCompiler.SHARED_OBJECT, objects,
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\_msvccompiler.py", line 460, in link
      lib_opts = gen_lib_options(self,
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\site-packages\setuptools\msvc.py", line 329, in msvc14_gen_lib_options
      return get_unpatched(msvc14_gen_lib_options)(*args, **kwargs)
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\ccompiler.py", line 1093, in gen_lib_options
      opt = compiler.runtime_library_dir_option(dir)
    File "C:\Users\a4z\AppData\Local\Programs\Python\Python39\lib\distutils\_msvccompiler.py", line 519, in runtime_library_dir_option
      raise DistutilsPlatformError(
  distutils.errors.DistutilsPlatformError: don't know how to set runtime library search path for MSVC
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'C:\Users\a4z\work\djinni\support-lib\build\py2\CMakeFiles 
\aadf18d4fcbdff197304b4b5d83cea85\pycffi_lib_build.py.rule;C:\Users\a4z\work\djinni\support-lib\build\py2\CMakeFiles\5ce445708c0c39eeaefe4c3c138abd3a\Foo_Record.py.rule;C:\Users\a4z\work\djinni\support-lib\build\py 
2\CMakeFiles\0723da9f51c0e0e7e15825be3e6cbffb\PyCFFIlib_cffi.rule' exited with code 1. [C:\Users\a4z\work\djinni\support-lib\build\py2\test-suite\PyCFFIlib_cffi.vcxproj]
~\work\djinni\support-lib [private/moreWindowsTweaks +4 ~1 -0 !] (18:36:58)
```

**Comment by a4z** [comment]
Closed in favor of #42
